get-terraform-provider-versions/get_provider_versions.py

```diff
@@ -0,0 +1,229 @@
+                    is_disabled: bool = r.get("disabled", False)
+
+                    if is_archived:
+                        logging.info(
+                            f"Skipping repository {name}, \
+because this repo has been archived."
+                        )
+                    elif is_disabled:
+                        logging.info(
+                            f"Skipping repository {name}, \
+because this repo has been disabled."
+                        )
+                    else:
+                        repo_has_hcl = repo.does_repo_contain_hcl(name)
+                        if repo_has_hcl:
+                            logging.info(
+                                f"Performing Terraform provider \
+analysis on the repository {name}."
+                            )
+
+                            f = tempfile.TemporaryDirectory()
+
+                            repository_name: str = r.get("name")
+
+                            clone_url: str = r.get("clone_url")
+                            clone_command: str = f"git clone {clone_url} {f.name} -q"
+                            logging.info(
+                                "\tCreating a local Clone of the GitHub repository."
+                            )
+                            subprocess.run(clone_command.split(" "), shell=True)
+                            logging.info("\tClone complete.")
+
+                            source_base: str = f.name
+                            for root, dirs, files in os.walk(source_base):
+                                for filename in files:
+                                    file_path: str = os.path.join(root, filename)
+                                    if (
+                                        file_path.find(".terragrunt-cache") == -1
+                                        and file_path.find(".git") == -1
+                                    ):
+                                        if file_path.endswith(".tf"):
+                                            used_providers: TerraformProviders = (
+                                                parse_terraform_file(
+                                                    f.name,
+                                                    repository_name,
+                                                    file_path,
+                                                    used_providers,
+                                                )
+                                            )
+                                        else:
+                                            continue
+                        else:
+                            logging.info(
+                                f"Skipping Terraform provider analysis \
+on the repository {name} because it does not \
+contain HCL code."
+                            )
+
+    used_providers.get_latest_versions()
+
+    if output_format == "csv":
+        out_str: str = "Repository Name,Terraform File,Provider Name"
```

Comment:
A more pythonic approach than string concatination is:

```
out_str: str = "Repository Name,Terraform File,Provider Name, " \
               "Version Used,Latest Version,Comment"
```

Reply:
Updated now to use the more pythonic approach :-)